Sprint 2 - Progress

KevinRebakure opened the pull request:
### Tasks completed

1. [(Admin) Add cohort Dropdown to all forms page](https://trello.com/c/GO29gUOA/107-admin-add-cohort-dropdown-to-all-forms-page)
2. [(Admin) Prevent Duplicate Forms Due to Unedited New Form](https://trello.com/c/IVDanRyv/97-admin-prevent-duplicate-forms-due-to-unedited-new-form)
3. [Make filter and sorting controls always visible ](https://trello.com/c/ls1UELys/110-make-filter-and-sorting-controls-always-visible)

KevinRebakure committed:
feat: add link to the join waitlist form

VictorMugisha committed:
feat: implement filtering on the forms page using dropdown

AL2002MI08 committed:
feat: enable navigation to external gym website

VictorMugisha committed:
feat: implemented the create form modal to add title and description before creating a form

AL2002MI08 committed:
Merge 'dev' into feat/open-external-gym-link

InshutiSouvede committed:
fix: validate new password and give correct message when no changes were made

VictorMugisha committed:
refactor: extracting onSubmit logic into a utility function

VictorMugisha committed:
refactor: moving interface in separate file and implementing other reviews

VictorMugisha committed:
Merge pull request #82 from kb21-rw/ft/filtering-forms

Implementing the filtering functionality

AL2002MI08 committed:
Merge pull request #86 from kb21-rw/staging

Form update on save

AL2002MI08 committed:
Merge pull request #87 from kb21-rw/staging

Fix form not updating on save

VictorMugisha committed:
merge: resolving conflict with dev branch

InshutiSouvede committed:
chore: remove unnecessary comment

InshutiSouvede committed:
Merge pull request #85 from kb21-rw/fix/update-password

Update Password

AL2002MI08 committed:
Merge pull request #84 from kb21-rw/feat/open-external-gym-link

Open "Learn More About The Gym" Link in New Tab

VictorMugisha committed:
Merge pull request #83 from kb21-rw/ft/create-form-modal

Create new form modal

AL2002MI08 committed:
refactor: remove navigation when form is created and add alert message for sucessful form creation

AL2002MI08 committed:
fix: prevent rename of forms to already exist form names

KevinRebakure committed:
refactor: file paths changed

KevinRebakure committed:
chore: install socketio-client

VictorMugisha committed:
feat: making filtering controls always visible on DataGrid components

KevinRebakure committed:
feat: create wait list socket context

KevinRebakure committed:
feat: Update the user after they have joined the waitlist

KevinRebakure committed:
refactor: remove unnecessary socketio initialisations

KevinRebakure committed:
refactor: change the message that's displayed to a user after joining the wait list

KevinRebakure committed:
feat: create the room

KevinRebakure committed:
refactor: rename the event

KevinRebakure committed:
feat: display accurate persisted data from the database

KevinRebakure committed:
feat: store a google scripts file

KevinRebakure committed:
refactor: put back the helper file that was taken out of utils directory by accident

KevinRebakure committed:
Merge branch 'dev' into feat/join-waitlist

InshutiSouvede committed:
fix: make sortIcon visible

AL2002MI08 committed:
refactor: add navigation to edit page when a form is created

AL2002MI08 committed:
fix: description validation when updating form

AL2002MI08 committed:
refactor: remove console logs

KevinRebakure committed:
feat: remove the join wait list button upon receiving the message

KevinRebakure committed:
Merge pull request #88 from kb21-rw/fix/duplicate-forms

Prevent the creation of duplicate forms

KevinRebakure committed:
Merge pull request #89 from kb21-rw/ft/display-filtering-controls

Make Filtering Controls Visible

KevinRebakure committed:
feat: add error handling and reconnecting

KevinRebakure committed:
feat: send an alert message to the user if they enter a wrong email

musayann commented on src/components/ui/CreateFormDropdown.tsx:
```diff
@@ -1,104 +1,76 @@
+import { PlusIcon } from "@radix-ui/react-icons"
+import { FormType } from "../../utils/types"
+import classNames from "classnames"
+import CreateForm from "../modals/CreateForm"
```
I would recommend in the future setting up a prettier and Eslint configuration and checks for prettier and Eslint in the CI (GitHub actions)

KevinRebakure committed:
refactor: rename variables properly

KevinRebakure committed:
refactor: refactor

KevinRebakure committed:
Merge branch 'feat/join-waitlist' into dev

dependabot[bot] committed:
chore(deps): bump axios from 1.7.9 to 1.8.2 (#96)

Bumps [axios](https://github.com/axios/axios) from 1.7.9 to 1.8.2.
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v1.x/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v1.7.9...v1.8.2)

---
updated-dependencies:
- dependency-name: axios
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump esbuild, @vitejs/plugin-react and vite (#97)

Bumps [esbuild](https://github.com/evanw/esbuild) to 0.25.1 and updates ancestor dependencies [esbuild](https://github.com/evanw/esbuild), [@vitejs/plugin-react](https://github.com/vitejs/vite-plugin-react/tree/HEAD/packages/plugin-react) and [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite). These dependencies need to be updated together.


Updates `esbuild` from 0.18.20 to 0.25.1
- [Release notes](https://github.com/evanw/esbuild/releases)
- [Changelog](https://github.com/evanw/esbuild/blob/main/CHANGELOG-2023.md)
- [Commits](https://github.com/evanw/esbuild/compare/v0.18.20...v0.25.1)

Updates `@vitejs/plugin-react` from 4.0.4 to 4.3.4
- [Release notes](https://github.com/vitejs/vite-plugin-react/releases)
- [Changelog](https://github.com/vitejs/vite-plugin-react/blob/main/packages/plugin-react/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite-plugin-react/commits/v4.3.4/packages/plugin-react)

Updates `vite` from 4.5.9 to 6.2.3
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v6.2.3/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v6.2.3/packages/vite)

---
updated-dependencies:
- dependency-name: esbuild
  dependency-type: indirect
- dependency-name: "@vitejs/plugin-react"
  dependency-type: direct:development
- dependency-name: vite
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

VictorMugisha committed:
refactor: formatting all files

KevinRebakure committed:
feat: use Controller component to prevent the search input from clearing when the user make a search

VictorMugisha committed:
chore: update the format script in the package.json

VictorMugisha committed:
test: testing pre-commit husky hook

VictorMugisha committed:
test: testing husky hooks in package.json

AL2002MI08 committed:
chore: install prettier plugins

AL2002MI08 committed:
chore: change checkmark icon

AL2002MI08 committed:
chore: change checkmark icon

AL2002MI08 committed:
Merge branch 'ft/precommit-issue' of https://github.com/kb21-rw/trainee-app-frontend into ft/precommit-issue

AL2002MI08 committed:
chore: resolve conflicts

AL2002MI08 committed:
feat: add GitHub actions workflow for prettier and ESLint checks

AL2002MI08 committed:
chore: fix yml config

AL2002MI08 committed:
refactor: eslint-prettier yml file

AL2002MI08 committed:
feat: add eslint-prettier yml file

AL2002MI08 committed:
chore: fix babel runtime issue

AL2002MI08 committed:
feat: update eslint yaml file

AL2002MI08 committed:
chore: trigger github actions

AL2002MI08 committed:
chore: update typescript version

AL2002MI08 committed:
fix: nodejs version issue

AL2002MI08 committed:
add: package-lock.json file

AL2002MI08 committed:
chore: remove unnecessary scripts

AL2002MI08 committed:
refactor: eslint prettier yml file

AL2002MI08 committed:
refactor: update eslint prettier yml file

AL2002MI08 committed:
Merge pull request #101 from kb21-rw/ft/precommit-issue

Add GitHub Actions Workflow for ESLint and Prettier Checks
